app/react/Layout/DocumentsList.js

```diff
@@ -44,6 +42,14 @@ class DocumentsList extends Component {
     }
   }
 
+  onEndScroll() {
+    const from = this.props.documents.get('rows').size;
+    const DEFAULT_PAGE_SIZE = 30;
+    if (from) {
+      this.loadMoreDocuments(DEFAULT_PAGE_SIZE, from);
+    }
+  }
+
```

Comment:
We had to relocate this function from TableViewer because it depends on the number of loaded entities, and since we introduce a new component for the rows in order to prevent the rerendering of the table at receiving the entities, this new component couldn't fire the scroll event on the table.
Actually, this method can be used for ListView too instead of the load more buttons.

Reply:
Well, you're right and I totally agree with you. We did this trying to balance the use of states within the components. While the ideal solution is what you propose, that means that the table view component calling this should know the amount of documents loadad (which doesn't currently, because the scroll is handled at a parent component and the rows are rendered at a child component), meaning that the table itself will re render when the rows change, which we were trying to avoid. Anyways, I think we can work it in another way. This handler is not necessarily a scroll handler but is about loading more. So we could rename it to "loadNextPage" (or simillar) which could be more descriptive, and could be easily used in future display strategies. In fact, it's a "loadmore" that doesn't require you to care about the current state (on the client side of the method.

Reply:
Ok.  Yeah, I can see the argument here.  Still, the value of documents loaded could be stored in the redux store and consumed down the road as well.  I'm OK with this.
I agree it *could* be used in DocumentsList, but not as implemented.

I agree it's a balanance: where should it be, what is store values and which are state, what is a subcomponent what gets a new file.  Yes.  That is what Konz meant by: "we don't have a unified way of doing things".  We should aim to have all these things somehow standardized. 

Still, I'm OK with this.